ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivity.kt

```diff
@@ -100,6 +100,14 @@ internal class PaywallActivity : ComponentActivity(), PaywallListener {
         finish()
     }
 
+    override fun onRestoreCompleted(customerInfo: CustomerInfo) {
+        setResult(RESULT_OK, createResultIntent(PaywallResult.Restored(customerInfo)))
+        val requiredEntitlementIdentifier = getArgs()?.requiredEntitlementIdentifier ?: return
+        if (customerInfo.entitlements.active.containsKey(requiredEntitlementIdentifier)) {
+            finish()
+        }
+    }
+
```

Comment:
hmmm this is making me think... if anyone is already doing this (finishing the activity when they get a `PaywallResult.Restored`, this will break their implementation. Right? Are we doing this in iOS?

Reply:
`PaywallResult.Restored` is a new sealed class element. Until now, I don't think users could detect whether the activity had restored or not.

As for iOS. I do think we have a `didFinishRestoringWith` delegate developers can use for this but I don't think it autocloses. That's something developers would need to handle themselves.

Reply:
😮‍💨 ok, should have kept reading

sorry for the noise

Reply:
I think we can autoclose and if anyone complains, we can add a setting somehow. I think autoclosing is a better default experience

Reply:
Well, in iOS I think we are not autoclosing for now. Not on a normal purchase nor on a restore. We talked about this before, and we felt it was more in line with iOS to not autodismiss. CC @aboedo @NachoSoto for confirmation.

But in android, since the developer doesn't have good access to the paywall activity, nor a way to commuinicate with it once the paywall activity is launched, we need to handle it for them.